src/coreclr/jit/codegencommon.cpp

```diff
@@ -6269,65 +6252,212 @@ void CodeGen::genZeroInitFrame(int untrLclHi, int untrLclLo, regNumber initReg,
             GetEmitter()->emitIns_R_I(INS_sub, EA_PTRSIZE, rCnt, 1, INS_FLAGS_SET);
-#else  // !TARGET_ARM
-            GetEmitter()->emitIns_R_R_R_I(INS_stp, EA_PTRSIZE, REG_ZR, REG_ZR, rAddr, 2 * REGSIZE_BYTES,
-                                          INS_OPTS_POST_INDEX); // zero stack slots
-            GetEmitter()->emitIns_R_R_I(INS_subs, EA_PTRSIZE, rCnt, rCnt, 1);
-#endif // !TARGET_ARM
             GetEmitter()->emitIns_J(INS_bhi, NULL, -3);
             uCntBytes %= REGSIZE_BYTES * 2;
         }
 
         if (uCntBytes >= REGSIZE_BYTES) // check and zero the last register-sized stack slot (odd number)
         {
-#ifdef TARGET_ARM
             GetEmitter()->emitIns_R_R_I(INS_str, EA_PTRSIZE, rZero1, rAddr, 0);
-#else  // TARGET_ARM
-            if ((uCntBytes - REGSIZE_BYTES) == 0)
+            uCntBytes -= REGSIZE_BYTES;
+        }
+
+        noway_assert(uCntBytes == 0);
+
+#elif defined(TARGET_ARM64)
+        int bytesToWrite = untrLclHi - untrLclLo;
+
+        const regNumber zeroSimdReg          = REG_ZERO_INIT_FRAME_SIMD;
+        bool            simdRegZeroed        = false;
+        const int       simdRegPairSizeBytes = 2 * FP_REGSIZE_BYTES;
+
+        regNumber addrReg = REG_ZERO_INIT_FRAME_REG1;
+
+        if (addrReg == initReg)
+        {
+            *pInitRegZeroed = false;
+        }
+
+        int addrOffset = 0;
+
+        // The following invariants are held below:
+        //
+        //   1) [addrReg, #addrOffset] points at a location where next chunk of zero bytes will be written;
+        //   2) bytesToWrite specifies the number of bytes on the frame to initialize;
+        //   3) if simdRegZeroed is true then 128-bit wide zeroSimdReg contains zeroes.
+
+        const int bytesUseZeroingLoop = 192;
+
+        if (bytesToWrite >= bytesUseZeroingLoop)
+        {
+            // Generates the following code:
+            //
+            // When the size of the region is greater than or equal to 256 bytes
+            // **and** DC ZVA instruction use is permitted
+            // **and** the instruction block size is configured to 64 bytes:
+            //
+            //    movi    v16.16b, #0
+            //    add     x9, fp, #(untrLclLo+64)
+            //    add     x10, fp, #(untrLclHi-64)
+            //    stp     q16, q16, [x9, #-64]
+            //    stp     q16, q16, [x9, #-32]
+            //    bfm     x9, xzr, #0, #5
+            //
+            // loop:
+            //    dc      zva, x9
+            //    add     x9, x9, #64
+            //    cmp     x9, x10
+            //    blo     loop
+            //
+            //    stp     q16, q16, [x10]
+            //    stp     q16, q16, [x10, #32]
+            //
+            // Otherwise:
+            //
+            //     movi    v16.16b, #0
+            //     add     x9, fp, #(untrLclLo-32)
+            //     mov     x10, #(bytesToWrite-64)
+            //
+            // loop:
```

Comment:
Did you verify if aligning the loop gives any benefits as @TamarChristinaArm  pointed?

Reply:
I did two experiments with aligning the head of a loop on 16-bytes and 32-bytes boundaries - neither demonstrated *measurable* performance differences, so I decided not to implement the alignment.